Case views cleanup

dannyroberts opened the pull request:
This pull request makes all references to `case/all_cases` go through dbaccessors. For each case I did this in one of three ways:
- move the function to a db accessors file and clean up its interface
- use an existing db accessor that does the same thing (may use a different view)
- delete the code

In the process I also got rid of the `CaseQueryMixin`, which I felt was too direct/generic a wrapper for that specific view, which doesn't really let you refactor the implementation. I replaced all usages of it with calls to functions whose names specifically reflect what they're being used for, so it's easy to swap out the backend if desired.

**Note**: there is one behavioral change that I'm aware of and that is that as an artifact of switching some view calls to use `hqcase/types_by_domain` instead of `case/all_cases`, cases with missing/null/empty user ids are now no longer excluded from the result. This doesn't worry me, that's a pretty rare edge case (you can't configure an app to do that on the app builder), and it's unclear why you would want to exclude such cases anyway.

dannyroberts committed:
get rid of one usage of case/all_cases

to use an existing function
that relies on hqcase/types_by_domain

dannyroberts committed:
add get_open_case_docs_by_type for call center

- refactor get_call_center_cases to use it
- test it in hqcase/tests/test_dbaccessors.py

dannyroberts committed:
remove add_case_properties cleanup mgmt cmd

rather than refactor its case/all_cases usage

dannyroberts committed:
replace CaseTypeMixin.get_case_types with get_case_types_for_domain

- test get_case_types_for_domain
- change it to use hqcase/types_by_domain
- remove CaseTypeMixin.get_case_types and replace usages

dannyroberts committed:
remove unused CaseTypeMixin method

- unused
- referenced case/all_cases, so one fewer reference

dannyroberts committed:
remove reassign_cases_to_correct_owner cleanup

- was for 2013 bihar migration
- one fewer call to CaseQueryMixin.get_all_cases

dannyroberts committed:
add get_cases_in_domain dbaccessor

- replace a CommCareCase.get_all_cases call
- add test
- refactor similar tests to be clearer

dannyroberts committed:
consolidate two identical function definitions

of get_cases_in_domain

dannyroberts committed:
remove code for "open cases browser"

that I wrote in 2011 and that was never exposed

dannyroberts committed:
add get_open_case_ids_in_domain

- rename get_open_case_docs_by_type to get_open_case_docs_in_domain
- have get_open_case_docs_in_domain call get_open_case_ids_in_domain
- add functionality to call get_open_case_(ids|docs)_in_domain without type
- rename parameter from case_type to type to match other dbaccessor functions

dannyroberts committed:
use dbaccessors in generate_case_export_payload

remove a call to CommCareCase.get_all_cases

dannyroberts committed:
use get_case_ids_in_domain in bihar_cleanup_case

- remove a CommCareCase.get_all_cases call

dannyroberts committed:
use dbaccessors in CommCareCaseResource

dannyroberts commented on corehq/apps/api/resources/v0_1.py:
```diff
@@ -312,10 +316,15 @@ def obj_get_list(self, bundle, **kwargs):
+
+        cases = imap(CommCareCase.wrap,
+                     iter_docs(CommCareCase.get_db(), case_ids))
+        return list(cases)
```
Does our case api really just read all of a domain's cases into memory!?

benrudolph replied:
yikes

dannyroberts committed:
have cloudcare API use dbaccessors

instead of CommCareCase.get_all_cases

dannyroberts committed:
use dbaccessors in opm docs

dannyroberts committed:
use dbaccessors in kill_cloudant mgmt cmd

dannyroberts committed:
remove CaseQueryMixin

it is now unused

dannyroberts committed:
use dbaccessors for cases in mvp_force_update

dimagimon commented on custom/_legacy/mvp/management/commands/mvp_force_update.py:
```diff
@@ -1,7 +1,9 @@
-from gevent import monkey;
+from gevent import monkey; monkey.patch_all()
```
E702 multiple statements on one line (semicolon)

dannyroberts committed:
use get_case_ids_in_domain in mvp_migrate_lesser_case_types_mar2015

dannyroberts committed:
use dbaccessors in bihar_cleanup_case

dannyroberts committed:
tweak test comment

dannyroberts commented:
(not sure about that first test failure, it doesn't happen locally, and I remember seeing and quickly fixing it in the course of working. I'll let the other tests finish before looking more)

benrudolph commented on corehq/apps/api/resources/v0_1.py:
```diff
@@ -312,10 +316,15 @@ def obj_get_list(self, bundle, **kwargs):
-        cases = CommCareCase.get_all_cases(domain, case_type=case_type, status=status, include_docs=True)
-        return list(cases)
 
+        if include_closed:
```
isn't this always `True` since it's statically defined above?

benrudolph replied:
nvm, just saw that it was being indexed in the actual file (the diff misled me)

benrudolph commented on custom/_legacy/mvp/management/commands/mvp_force_update.py:
```diff
@@ -72,20 +74,21 @@ def update_indicators_for_xmlns(self, domain, form_label_filter=None):
+                "reports_forms/all_forms",
                 reduce=False,
-                include_docs=True,
+                include_docs=False,
```
:+1:

benrudolph commented:
really nice cleanup. don't have too much context on what you can and cannot delete, but i think the move to dbaccessors and getting rid of `CaseQueryMixin` is sweet. test failure looks real

```
======================================================================
FAIL: testGetAllIdsOnly (corehq.apps.cloudcare.tests.test_api.CaseAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/cloudcare/tests/test_api.py", line 178, in testGetAllIdsOnly
    self.assertEqual(self.expectedAll, len(list))
AssertionError: 12 != 14
```

dannyroberts committed:
fix bad interaction between tests

Version2CaseParsingTest left cases around that interfered with CaseAPITest
this only got exposed as a problem when get_filtered_cases switched to
use hqcase/types_by_domain, and as a side effect no longer filters out
cases missing user_id. This edge case is not likely to be something that
matters, as all cases should have a user_id. (Version2CaseParsingTest
happened to leave around ones that didn't.)

dannyroberts commented:
:white_check_mark:

dannyroberts commented:
This is good to review/merge on my end
